### DISCUSSION TITLE
coal retirement

### DISCUSSION THREAD

**Discussion by ssattler** [NREL/ReEDS-2.0#154] [Q&A: Analysis]
What is a good way to write a constraint to require the coal fleet to retire by a specific year?

**Comment by wesleyjcole** [comment]
There are several ways this could be done.  Here are a few:

- The plant database (inputs/capacitydata/ReEDS_generator_database_final_EIA-NEMS.csv) has the retirement dates for all of the power plants.  You could modify the dates in that file.  If you wanted the dates controlled by a switch, you could add a switch that gives a suffix for that database file, and then have multiple versions, each with a different retirement date.
- You can use the forced_retirements inputs (inputs/state_policies/forced_retirements.csv).  It lists the year that a given technology in a given region must retire.
- You can adjust the valcap parameter after is has been populated to exclude coal technologies after a given year: `valcap(i,v,r,t)$[coal(i)$(yeart(t) >= year_with_no_more_coal)] = no ;`
- You can add a constraint to the model that says coal capacity equals zero in a given year `CAP(i,v,r,t) =e= 0 ; ` and just make sure the equation applied to only coal techs and only after the year you desire.

**Reply by ssattler** [reply]
Thanks! Is there a way that the model can decide which year individual coal plants retire, instead of prescribing a retirement year? Would either of the last 2 methods allow the model to retire coal over a period instead of forcing the shutdown in a single time step?

**Reply by wesleyjcole** [reply]
The model already allows for endogenous retirements after the year set be GSw_RetireYear, so any of these methods would still allow coal to retire before the final period that you set.  If you want to reduce coal capcity gradually over time, then you could use the last option I listed, but have the right-hand side be a parameter (max_coal_capacity(t)) that specifies how much coal capacity is allowed in each year.
